core/src/main/java/tc/oc/pgm/filters/FilterParser.java

```diff
@@ -470,6 +471,18 @@ public WearingItemFilter parseWearingItem(Element el) throws InvalidXMLException
     return new WearingItemFilter(factory.getKits().parseRequiredItem(el));
   }
 
+  @MethodParser("effect")
+  public HasEffectFilter parseHasEffect(Element el) throws InvalidXMLException {
+    Duration minDuration = XMLUtils.parseDuration(Node.fromAttr(el, "min-duration"));
+    Duration maxDuration = XMLUtils.parseDuration(Node.fromAttr(el, "max-duration"));
```

Comment:
Create a Range.atMost, Range.atLeast, or Range.closed based on if the durations are present or not, and use that in the filter